### ISSUE TITLE
[Firestore] Xcode 14.3 Analyzer catches issues with Firestore 

### ISSUE THREAD

**Issue by ncooke3** [firebase/firebase-ios-sdk#11240]
### Description

Xcode 14.3 is picking up two issues in Firestore's source and two issues in abseil. I'm not sure about addressing the abseil ones, but the Firestore issues can be addressed:

```
    - WARN  | [iOS] xcodebuild:  /Users/nickcooke/Developer/firebase-ios-sdk/Firestore/core/src/model/mutation_batch.cc:100:7: warning: Method called on moved-from object 'mutated_fields' [cplusplus.Move]
    - WARN  | [iOS] xcodebuild:  /Users/nickcooke/Developer/firebase-ios-sdk/Firestore/core/src/model/mutation_batch.cc:109:7: warning: Method called on moved-from object 'mutated_fields' [cplusplus.Move]
```

### Reproducing the issue

```
sudo xcode-select -s /Applications/Xcode_14.3.app/Contents/Developer 
bundle exec pod lib lint --sources=https://github.com/firebase/SpecsDev.git,https://github.com/firebase/SpecsStaging.git,https://cdn.cocoapods.org/ --include-podspecs=\{FirebaseCore.podspec,FirebaseCoreInternal.podspec\} --analyze FirebaseFirestore.podspec --analyze --platforms=ios --skip-tests

Still working, running for 1min.
Still working, running for 2min.
Still working, running for 3min.
Still working, running for 4min.
Still working, running for 5min.
Still working, running for 6min.
Still working, running for 7min.
Still working, running for 8min.
Still working, running for 9min.
Still working, running for 10min.
Still working, running for 11min.
Still working, running for 12min.
Still working, running for 13min.
Still working, running for 14min.
Still working, running for 15min.
Still working, running for 16min.
Still working, running for 17min.
Still working, running for 18min.
Still working, running for 19min.
Still working, running for 20min.
Still working, running for 21min.
Still working, running for 22min.
Still working, running for 23min.
Still working, running for 24min.
Still working, running for 25min.
 -> FirebaseFirestore (10.10.0)
    - WARN  | compiler_flags: Warnings must not be disabled(`-Wno compiler` flags).
    - ERROR | [iOS] build_pod: Static Analysis failed. You can use `--verbose` for more information. You can use `--no-clean` to save a reproducible buid environment.
    - NOTE  | xcodebuild:  note: Using codesigning identity override: -
    - NOTE  | [iOS] xcodebuild:  note: Building targets in dependency order
    - NOTE  | [iOS] xcodebuild:  note: Metadata extraction skipped. No AppIntents.framework dependency found. (in target 'FirebaseCoreInternal' from project 'Pods')
    - WARN  | xcodebuild:  /Users/nickcooke/Library/Developer/Xcode/DerivedData/App-haffuhdpiyydpocoaqnlykxrjjko/Build/Products/Release-iphonesimulator/abseil/absl.framework/Headers/strings/str_split.h:516:7: warning: Forwarding reference passed to std::move(), which may unexpectedly cause lvalues to be moved; use std::forward() instead [bugprone-move-forwarding-reference]
    - WARN  | xcodebuild:  /Users/nickcooke/Library/Developer/Xcode/DerivedData/App-haffuhdpiyydpocoaqnlykxrjjko/Build/Products/Release-iphonesimulator/abseil/absl.framework/Headers/strings/str_split.h:541:7: warning: Forwarding reference passed to std::move(), which may unexpectedly cause lvalues to be moved; use std::forward() instead [bugprone-move-forwarding-reference]
    - WARN  | [iOS] xcodebuild:  /Users/nickcooke/Developer/firebase-ios-sdk/Firestore/core/src/model/mutation_batch.cc:100:7: warning: Method called on moved-from object 'mutated_fields' [cplusplus.Move]
    - WARN  | [iOS] xcodebuild:  /Users/nickcooke/Developer/firebase-ios-sdk/Firestore/core/src/model/mutation_batch.cc:109:7: warning: Method called on moved-from object 'mutated_fields' [cplusplus.Move]
    - NOTE  | [iOS] xcodebuild:  note: Metadata extraction skipped. No AppIntents.framework dependency found. (in target 'App' from project 'App')
    - NOTE  | [iOS] xcodebuild:  Pods.xcodeproj: warning: The iOS Simulator deployment target 'IPHONEOS_DEPLOYMENT_TARGET' is set to 9.0, but the range of supported deployment target versions is 11.0 to 16.4.99. (in target 'nanopb' from project 'Pods')
    - NOTE  | [iOS] xcodebuild:  Pods.xcodeproj: warning: The iOS Simulator deployment target 'IPHONEOS_DEPLOYMENT_TARGET' is set to 9.0, but the range of supported deployment target versions is 11.0 to 16.4.99. (in target 'gRPC-Core' from project 'Pods')
    - NOTE  | [iOS] xcodebuild:  warning: Run script build phase 'Create Symlinks to Header Folders' will be run during every build because it does not specify any outputs. To address this warning, either add output dependencies to the script phase, or configure it to run in every build by unchecking "Based on dependency analysis" in the script phase. (in target 'gRPC-Core' from project 'Pods')
    - NOTE  | [iOS] xcodebuild:  Pods.xcodeproj: warning: The iOS Simulator deployment target 'IPHONEOS_DEPLOYMENT_TARGET' is set to 9.0, but the range of supported deployment target versions is 11.0 to 16.4.99. (in target 'gRPC-C++-gRPCCertificates-Cpp' from project 'Pods')
    - NOTE  | [iOS] xcodebuild:  Pods.xcodeproj: warning: The iOS Simulator deployment target 'IPHONEOS_DEPLOYMENT_TARGET' is set to 9.0, but the range of supported deployment target versions is 11.0 to 16.4.99. (in target 'gRPC-C++' from project 'Pods')
    - NOTE  | [iOS] xcodebuild:  warning: Run script build phase 'Create Symlinks to Header Folders' will be run during every build because it does not specify any outputs. To address this warning, either add output dependencies to the script phase, or configure it to run in every build by unchecking "Based on dependency analysis" in the script phase. (in target 'gRPC-C++' from project 'Pods')
    - NOTE  | [iOS] xcodebuild:  Pods.xcodeproj: warning: The iOS Simulator deployment target 'IPHONEOS_DEPLOYMENT_TARGET' is set to 9.0, but the range of supported deployment target versions is 11.0 to 16.4.99. (in target 'abseil' from project 'Pods')
    - NOTE  | [iOS] xcodebuild:  warning: Run script build phase 'Create Symlinks to Header Folders' will be run during every build because it does not specify any outputs. To address this warning, either add output dependencies to the script phase, or configure it to run in every build by unchecking "Based on dependency analysis" in the script phase. (in target 'abseil' from project 'Pods')
    - NOTE  | [iOS] xcodebuild:  Pods.xcodeproj: warning: The iOS Simulator deployment target 'IPHONEOS_DEPLOYMENT_TARGET' is set to 9.0, but the range of supported deployment target versions is 11.0 to 16.4.99. (in target 'PromisesObjC' from project 'Pods')
    - NOTE  | [iOS] xcodebuild:  Pods.xcodeproj: warning: The iOS Simulator deployment target 'IPHONEOS_DEPLOYMENT_TARGET' is set to 9.0, but the range of supported deployment target versions is 11.0 to 16.4.99. (in target 'GoogleUtilities' from project 'Pods')
    - NOTE  | [iOS] xcodebuild:  Pods.xcodeproj: warning: The iOS Simulator deployment target 'IPHONEOS_DEPLOYMENT_TARGET' is set to 10.0, but the range of supported deployment target versions is 11.0 to 16.4.99. (in target 'FirebaseCoreInternal' from project 'Pods')
    - NOTE  | [iOS] xcodebuild:  Pods.xcodeproj: warning: The iOS Simulator deployment target 'IPHONEOS_DEPLOYMENT_TARGET' is set to 10.0, but the range of supported deployment target versions is 11.0 to 16.4.99. (in target 'FirebaseCore' from project 'Pods')
    - NOTE  | [iOS] xcodebuild:  Pods.xcodeproj: warning: The iOS Simulator deployment target 'IPHONEOS_DEPLOYMENT_TARGET' is set to 9.0, but the range of supported deployment target versions is 11.0 to 16.4.99. (in target 'BoringSSL-GRPC' from project 'Pods')
    - NOTE  | [iOS] xcodebuild:  warning: Run script build phase 'Create Symlinks to Header Folders' will be run during every build because it does not specify any outputs. To address this warning, either add output dependencies to the script phase, or configure it to run in every build by unchecking "Based on dependency analysis" in the script phase. (in target 'BoringSSL-GRPC' from project 'Pods')

[!] FirebaseFirestore did not pass validation, due to 1 error and 5 warnings.
You can use the `--no-clean` option to inspect any issue.
```

### Firebase SDK Version

10.9.0

### Xcode Version

14.3

### Installation Method

N/A

### Firebase Product(s)

Firestore

### Targeted Platforms

All

### Relevant Log Output

```shell
Still working, running for 1min.
Still working, running for 2min.
Still working, running for 3min.
Still working, running for 4min.
Still working, running for 5min.
Still working, running for 6min.
Still working, running for 7min.
Still working, running for 8min.
Still working, running for 9min.
Still working, running for 10min.
Still working, running for 11min.
Still working, running for 12min.
Still working, running for 13min.
Still working, running for 14min.
Still working, running for 15min.
Still working, running for 16min.
Still working, running for 17min.
Still working, running for 18min.
Still working, running for 19min.
Still working, running for 20min.
Still working, running for 21min.
Still working, running for 22min.
Still working, running for 23min.
Still working, running for 24min.
Still working, running for 25min.
 -> FirebaseFirestore (10.10.0)
    - WARN  | compiler_flags: Warnings must not be disabled(`-Wno compiler` flags).
    - ERROR | [iOS] build_pod: Static Analysis failed. You can use `--verbose` for more information. You can use `--no-clean` to save a reproducible buid environment.
    - NOTE  | xcodebuild:  note: Using codesigning identity override: -
    - NOTE  | [iOS] xcodebuild:  note: Building targets in dependency order
    - NOTE  | [iOS] xcodebuild:  note: Metadata extraction skipped. No AppIntents.framework dependency found. (in target 'FirebaseCoreInternal' from project 'Pods')
    - WARN  | xcodebuild:  /Users/nickcooke/Library/Developer/Xcode/DerivedData/App-haffuhdpiyydpocoaqnlykxrjjko/Build/Products/Release-iphonesimulator/abseil/absl.framework/Headers/strings/str_split.h:516:7: warning: Forwarding reference passed to std::move(), which may unexpectedly cause lvalues to be moved; use std::forward() instead [bugprone-move-forwarding-reference]
    - WARN  | xcodebuild:  /Users/nickcooke/Library/Developer/Xcode/DerivedData/App-haffuhdpiyydpocoaqnlykxrjjko/Build/Products/Release-iphonesimulator/abseil/absl.framework/Headers/strings/str_split.h:541:7: warning: Forwarding reference passed to std::move(), which may unexpectedly cause lvalues to be moved; use std::forward() instead [bugprone-move-forwarding-reference]
    - WARN  | [iOS] xcodebuild:  /Users/nickcooke/Developer/firebase-ios-sdk/Firestore/core/src/model/mutation_batch.cc:100:7: warning: Method called on moved-from object 'mutated_fields' [cplusplus.Move]
    - WARN  | [iOS] xcodebuild:  /Users/nickcooke/Developer/firebase-ios-sdk/Firestore/core/src/model/mutation_batch.cc:109:7: warning: Method called on moved-from object 'mutated_fields' [cplusplus.Move]
    - NOTE  | [iOS] xcodebuild:  note: Metadata extraction skipped. No AppIntents.framework dependency found. (in target 'App' from project 'App')
    - NOTE  | [iOS] xcodebuild:  Pods.xcodeproj: warning: The iOS Simulator deployment target 'IPHONEOS_DEPLOYMENT_TARGET' is set to 9.0, but the range of supported deployment target versions is 11.0 to 16.4.99. (in target 'nanopb' from project 'Pods')
    - NOTE  | [iOS] xcodebuild:  Pods.xcodeproj: warning: The iOS Simulator deployment target 'IPHONEOS_DEPLOYMENT_TARGET' is set to 9.0, but the range of supported deployment target versions is 11.0 to 16.4.99. (in target 'gRPC-Core' from project 'Pods')
    - NOTE  | [iOS] xcodebuild:  warning: Run script build phase 'Create Symlinks to Header Folders' will be run during every build because it does not specify any outputs. To address this warning, either add output dependencies to the script phase, or configure it to run in every build by unchecking "Based on dependency analysis" in the script phase. (in target 'gRPC-Core' from project 'Pods')
    - NOTE  | [iOS] xcodebuild:  Pods.xcodeproj: warning: The iOS Simulator deployment target 'IPHONEOS_DEPLOYMENT_TARGET' is set to 9.0, but the range of supported deployment target versions is 11.0 to 16.4.99. (in target 'gRPC-C++-gRPCCertificates-Cpp' from project 'Pods')
    - NOTE  | [iOS] xcodebuild:  Pods.xcodeproj: warning: The iOS Simulator deployment target 'IPHONEOS_DEPLOYMENT_TARGET' is set to 9.0, but the range of supported deployment target versions is 11.0 to 16.4.99. (in target 'gRPC-C++' from project 'Pods')
    - NOTE  | [iOS] xcodebuild:  warning: Run script build phase 'Create Symlinks to Header Folders' will be run during every build because it does not specify any outputs. To address this warning, either add output dependencies to the script phase, or configure it to run in every build by unchecking "Based on dependency analysis" in the script phase. (in target 'gRPC-C++' from project 'Pods')
    - NOTE  | [iOS] xcodebuild:  Pods.xcodeproj: warning: The iOS Simulator deployment target 'IPHONEOS_DEPLOYMENT_TARGET' is set to 9.0, but the range of supported deployment target versions is 11.0 to 16.4.99. (in target 'abseil' from project 'Pods')
    - NOTE  | [iOS] xcodebuild:  warning: Run script build phase 'Create Symlinks to Header Folders' will be run during every build because it does not specify any outputs. To address this warning, either add output dependencies to the script phase, or configure it to run in every build by unchecking "Based on dependency analysis" in the script phase. (in target 'abseil' from project 'Pods')
    - NOTE  | [iOS] xcodebuild:  Pods.xcodeproj: warning: The iOS Simulator deployment target 'IPHONEOS_DEPLOYMENT_TARGET' is set to 9.0, but the range of supported deployment target versions is 11.0 to 16.4.99. (in target 'PromisesObjC' from project 'Pods')
    - NOTE  | [iOS] xcodebuild:  Pods.xcodeproj: warning: The iOS Simulator deployment target 'IPHONEOS_DEPLOYMENT_TARGET' is set to 9.0, but the range of supported deployment target versions is 11.0 to 16.4.99. (in target 'GoogleUtilities' from project 'Pods')
    - NOTE  | [iOS] xcodebuild:  Pods.xcodeproj: warning: The iOS Simulator deployment target 'IPHONEOS_DEPLOYMENT_TARGET' is set to 10.0, but the range of supported deployment target versions is 11.0 to 16.4.99. (in target 'FirebaseCoreInternal' from project 'Pods')
    - NOTE  | [iOS] xcodebuild:  Pods.xcodeproj: warning: The iOS Simulator deployment target 'IPHONEOS_DEPLOYMENT_TARGET' is set to 10.0, but the range of supported deployment target versions is 11.0 to 16.4.99. (in target 'FirebaseCore' from project 'Pods')
    - NOTE  | [iOS] xcodebuild:  Pods.xcodeproj: warning: The iOS Simulator deployment target 'IPHONEOS_DEPLOYMENT_TARGET' is set to 9.0, but the range of supported deployment target versions is 11.0 to 16.4.99. (in target 'BoringSSL-GRPC' from project 'Pods')
    - NOTE  | [iOS] xcodebuild:  warning: Run script build phase 'Create Symlinks to Header Folders' will be run during every build because it does not specify any outputs. To address this warning, either add output dependencies to the script phase, or configure it to run in every build by unchecking "Based on dependency analysis" in the script phase. (in target 'BoringSSL-GRPC' from project 'Pods')

[!] FirebaseFirestore did not pass validation, due to 1 error and 5 warnings.
You can use the `--no-clean` option to inspect any issue.
```

### If using Swift Package Manager, the project's Package.resolved

<details>
<summary>Expand <code>Package.resolved</code> snippet</summary>
<br>

```json

Replace this line with the contents of your Package.resolved.

```

</details>

### If using CocoaPods, the project's Podfile.lock

<details>
<summary>Expand <code>Podfile.lock</code> snippet</summary>
<br>

```yml

Replace this line with the contents of your Podfile.lock!

```

</details>

**Comment by ncooke3** [comment]
Hi @wu-hui, is this something you could take a look at? I suspect the fix is simple–– I just don't know enough C++ to understand what to do.

**Comment by wu-hui** [comment]
Sure, thanks for checking!

**Comment by ncooke3** [comment]
Closed via #11256
